### PR TITLE
New electron version

### PR DESCRIPTION
**Pull request by sdetweil**
* Does the pull request solve a **related** issue?
#1800 

change the electron version required

### CODE REVIEW

**Comment by sdetweil** [comment]
@MichMich how do I fix this build error? where package-lock.json has the old electron version number in it?
we should never save the package-lock file anyhow

**Comment by johankvint** [comment]
> @MichMich how do I fix this build error? where package-lock.json has the old electron version number in it?
> we should never save the package-lock file anyhow

Remove your local package-lock.json and run `npm install`. It should generate a new lock file with your dependencies

**Comment by sdetweil** [comment]
And then I check that new package-ock file. Duh. I hadn't don't nom install, just wanted to change the text file

**Comment by sdetweil** [comment]
so, what is the problem here ? all I did was npm install, and the commit the package.json file

File: package.json [hunk 1]
```diff
@@ -56,7 +56,7 @@
     "time-grunt": "latest"
   },
   "optionalDependencies": {
-    "electron": "^3.0.13"
+    "electron": "^6.0.12"
```
Comment (chadasapp) [inline package.json]: 6.0.x isn't supported anymore? I think it should be 6.1.x, if not 7.1.x

**Comment by sdetweil** [comment]
where is the comment about 6.1/7.1 vs 6.0.12? says it should be here, but I can't find it

we  WERE at 3.0.13... so its a big jump.. there are side effects.. as armv6l is no longer supported in 6.0.12.. we haven't tested beyond 6.0.12...  but we need to upgrade to fix other problems

**Comment by rvoitenko** [comment]
I have tested setup with electron 6.x and 7.x. The latest 6.1.5 works just fine, but not the 7.x
The platform was running debian buster and node 12.x

**Comment by sdetweil** [comment]
@rvoitenko   thanks.. as I suspected..

**Comment by sdetweil** [comment]
what api.js is this error from?

```
  1) Electron app environment
       "before each" hook for "should open a browserwindow":
     unknown error: Cannot read property 'name' of undefined
  Error: An unknown server-side error occurred while processing the command.
      at execute(<Function>, "require") - api.js:73:26
```

this sounds like the 
		webPreferences: {
			**nodeIntegration**: false,  (which should be true to enable require to work)

**Comment by ezar** [comment]
Can you apply?

**Comment by sdetweil** [comment]
@ezar what does that mean?

**Comment by MichMich** [comment]
@sdetweil Unfortunately I had to revert this PR. This was the one causing all the issues. I'll make an issue that we need to update Electron, but Travis needs to pass before we can.

If anyone want to run buster, they can always manually install the updated electron version, using:

```
npm install electron@6.0.12
```

**Comment by sdetweil** [comment]
It all rubs ok locally. So something to do with Travis setup. Not electron problem

**Comment by MichMich** [comment]
Absolutely. Electron probably gives issues with the used bistro on Travis. But we need Travis to release a new version and for me to merge PR's. So for now, this is the best option.

**Comment by sdetweil** [comment]
Yeh. We need instructions on how to update the Travis process as we enhance the system

**Comment by sdetweil** [comment]
Please reopen my electron  version issue

**Pull request merged**